modules/metrics/src/main/java/com/spotify/apollo/metrics/semantic/SemanticRequestMetrics.java

```diff
@@ -61,60 +86,124 @@
+                  "what", DROPPED_REQUEST_RATE.what(),
+                  "unit", "request"
+              )));
+    } else {
+      droppedRequests = Optional.empty();
+    }
 
     this.sentReplies = requireNonNull(sentReplies);
     this.sentErrors = requireNonNull(sentErrors);
 
-    registerRatioGauge(metricId, "1m", () -> RatioGauge.Ratio.of(this.sentErrors.getOneMinuteRate(),
-                                                                 this.sentReplies.getOneMinuteRate()));
-    registerRatioGauge(metricId, "5m", () -> RatioGauge.Ratio.of(this.sentErrors.getFiveMinuteRate(),
-                                                                 this.sentReplies.getFiveMinuteRate()));
-    registerRatioGauge(metricId, "15m", () -> RatioGauge.Ratio.of(this.sentErrors.getFifteenMinuteRate(),
-                                                                  this.sentReplies.getFifteenMinuteRate()));
+    registerRatioGauge(metricId, "1m", () -> Ratio.of(this.sentErrors.getOneMinuteRate(),
+                                                      this.sentReplies.getOneMinuteRate()));
+    registerRatioGauge(metricId, "5m", () -> Ratio.of(this.sentErrors.getFiveMinuteRate(),
+                                                      this.sentReplies.getFiveMinuteRate()));
+    registerRatioGauge(metricId, "15m", () -> Ratio.of(this.sentErrors.getFifteenMinuteRate(),
+                                                       this.sentReplies.getFifteenMinuteRate()));
   }
 
   private void registerRatioGauge(MetricId metricId,
                                   String stat,
-                                  Supplier<RatioGauge.Ratio> ratioSupplier) {
-    metricRegistry.register(
-        metricId.tagged("what", "error-ratio", "stat", stat),
-        new RatioGauge() {
-          @Override
-          protected Ratio getRatio() {
-            return ratioSupplier.get();
-          }
-        });
+                                  Supplier<Ratio> ratioSupplier) {
+    if (enabledMetrics.contains(ERROR_RATIO)) {
+      metricRegistry.register(
+          metricId.tagged("what", ERROR_RATIO.what(), "stat", stat),
+          new RatioGauge() {
+            @Override
+            protected Ratio getRatio() {
+              return ratioSupplier.get();
+            }
+          });
+    }
+  }
+
+  @Override
+  public void incoming(Request request) {
+    requestSizeHistogram
+        .ifPresent(histogram -> request.payload()
+            .ifPresent(payload -> histogram.update(payload.size())));
   }
 
   @Override
   public void fanout(int requests) {
-    fanoutHistogram.update(requests);
+    fanoutHistogram.ifPresent(histogram -> histogram.update(requests));
   }
 
   @Override
-  public void responseStatus(StatusType status) {
-    metricRegistry.meter(countRequestId.tagged("status-code", String.valueOf(status.code()))).mark();
+  public void response(Response<ByteString> response) {
+    if (enabledMetrics.contains(ENDPOINT_REQUEST_RATE)) {
```

Comment:
Keep Consumer as member instead of SemanticMetricRegistry and perform check once in constructor